src/org/pentaho/reporting/platform/plugin/BackgroundJobReportContentGenerator.java

```diff
@@ -87,23 +92,26 @@ public void createReportContent( final Serializable fileId, final String path,
     reportComponent.setPaginateOutput( true );
     reportComponent.setForceDefaultOutputTarget( false );
     reportComponent.setDefaultOutputTarget( HtmlTableModule.TABLE_HTML_PAGE_EXPORT_TYPE );
-    if ( path.endsWith( ".prpti" ) ) {
+    if ( path.endsWith( PRPTI ) ) {
       reportComponent.setForceUnlockPreferredOutput( true );
     }
     reportComponent.setInputs( inputs );
 
-    AsyncJobFileStagingHandler handler = new AsyncJobFileStagingHandler( userSession );
+    final AsyncJobFileStagingHandler handler = new AsyncJobFileStagingHandler( userSession );
     // will write to async stage target
     reportComponent.setOutputStream( handler.getStagingOutputStream() );
 
-    PentahoAsyncReportExecution
+    final PentahoAsyncReportExecution
       asyncExec = new PentahoAsyncReportExecution( path, reportComponent, handler, userSession, instanceId, audit );
 
-    IPentahoAsyncExecutor executor =
+    final IPentahoAsyncExecutor executor =
       PentahoSystem.get( PentahoAsyncExecutor.class, PentahoAsyncExecutor.BEAN_NAME, null );
     // delegation
     if ( reportComponent.validate() ) {
-      UUID uuid = executor.addTask( asyncExec, userSession );
+      final String precomputedId =
+        getRequestParameters().getStringParameter( PRECOMPUTED_ID, UUIDUtil.getUUIDAsString() );
```

Comment:
UUIDUtil is the most useless class ever. Whats wrong with 

    java.util.UUID.randomUUID().toString();